internal/provider/resource_gitlab_runner_test.go

```diff
@@ -0,0 +1,121 @@
+package provider
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/xanzy/go-gitlab"
+)
+
+func TestAccGitlabRunner_basic(t *testing.T) {
+	testAccCheck(t)
+
+	group := testAccCreateGroups(t, 1)[0]
+	//The runner token is not populated on the return from the group create, so re-retrieve it to get the token.
+	group, _, err := testGitlabClient.Groups.GetGroup(group.ID, &gitlab.GetGroupOptions{})
+	if err != nil {
+		t.Fail()
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() {},
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckManagedLicenseDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(`
+				resource "gitlab_runner" "this" {
+					token = "%s"
+					description = "Lorem Ipsum"
+				}
+				`, group.RunnersToken),
+			},
+			{
+				ResourceName:      "gitlab_runner.this",
+				ImportState:       true,
+				ImportStateVerify: true,
+				//These need to be ignored since they don't come back in the "get" command
+				ImportStateVerifyIgnore: []string{"authentication_token", "token"},
+			},
+			{
+				Config: fmt.Sprintf(`
+				resource "gitlab_runner" "this" {
+					token = "%s"
+					description = "Lorem Ipsum Dolor Sit Amet"
+				}
+				`, group.RunnersToken),
+			},
+			{
+				ResourceName:      "gitlab_runner.this",
+				ImportState:       true,
+				ImportStateVerify: true,
+				//These need to be ignored since they don't come back in the "get" command
+				ImportStateVerifyIgnore: []string{"authentication_token", "token"},
+			},
+		},
+	})
+}
+
+func TestAccGitlabRunner_comprehensive(t *testing.T) {
+	testAccCheck(t)
+
+	group := testAccCreateGroups(t, 1)[0]
+	//The runner token is not populated on the return from the group create, so re-retrieve it to get the token.
+	group, _, err := testGitlabClient.Groups.GetGroup(group.ID, &gitlab.GetGroupOptions{})
+	if err != nil {
+		t.Fail()
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() {},
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckManagedLicenseDestroy,
```

Comment:
also a wrong reference - you have to implement that check method in this test file yourself specific to the `gitlab_runner` resource :)

Reply:
Yeah... like you said, silly copy/paste mistake 😄